### PR TITLE
Documentation update

### PR DESCRIPTION
**Pull request by mmccall**
#### What does this PR do?

Updates the README to better instruct users on the difference between this code base and the various forks, and adds instructions for using the mapping generation tools. Also adds in a change-log.

#### Related JIRA tickets:

None, part of punch-list.

#### How should this be manually tested?

Review the instructions, ensure they are logical and that all the links work.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,5 +1,5 @@
 <!--
-Copyright (C) 2016 SRDC Yazilim Arastirma ve Gelistirme ve Danismanlik Tic. A.S.
+Copyright (C) 2019 Amida Technoloy Solutions, Inc.
```
Comment (au2) [inline README.md:3]: Misspelled Technology

File: README.md [hunk 2]
```diff
@@ -19,24 +19,51 @@ cda2fhir
...
+
+In addition to the above mappings, cda2fhir also uses and supports the Patient, Practitioner, PractitionerRole, Organization, Device, DocumentReference, Composition, and Provenance FHIR Resources.
+
+**This version of the cda2fhir library implements several additional features not present in the SRDC version. These include:**
+
+* cda2fhir is now capable of generating "transactional" bundles.
+* cda2fhir now supports the generation of Provenance objects, optionally taking in an Identifier resource and string representation of the source file to generate the accompanying Device and DocumentReference resources respectively.
+* Bundles now de-duplicate against themselves for certain resources; this is done to prevent duplicate resources from being created in a FHIR server, and works together with the ifNoneExist parameters.
+  * Medications are de-duplicated based on their encoding.
+  * Organizations and Practitioners are deduplicated based on identifier.
```
Comment (au2) [inline README.md]: Also organization identifier for medications.

File: README.md [hunk 3]
```diff
@@ -19,24 +19,51 @@ cda2fhir
...
+|Procedures           |Procedure|
+|Immunizations |Immunization, Medication|
+|Results| DiagnosticReport, Observation|
+|Vital Signs| Observation|
+|Problems (Conditions)| Condition|
+|Allergies and Intolerances| AllergyIntolerance|
+|Encounters| Encounter|
+
+In addition to the above mappings, cda2fhir also uses and supports the Patient, Practitioner, PractitionerRole, Organization, Device, DocumentReference, Composition, and Provenance FHIR Resources.
+
```
Comment (au2) [inline README.md:49]: Location as well.

File: README.md [hunk 4]
```diff
@@ -19,24 +19,51 @@ cda2fhir
...
+**This version of the cda2fhir library implements several additional features not present in the SRDC version. These include:**
+
+* cda2fhir is now capable of generating "transactional" bundles.
+* cda2fhir now supports the generation of Provenance objects, optionally taking in an Identifier resource and string representation of the source file to generate the accompanying Device and DocumentReference resources respectively.
+* Bundles now de-duplicate against themselves for certain resources; this is done to prevent duplicate resources from being created in a FHIR server, and works together with the ifNoneExist parameters.
+  * Medications are de-duplicated based on their encoding.
+  * Organizations and Practitioners are deduplicated based on identifier.
+* Bundles now use the "ifNoneExist" parameter to prevent duplicate resources from being created on a target FHIR server. This parameter uses the identifier field to prevent duplicates for all resources, with the exception of:
+  * Medications which is de-duplicated based on encoding, and
+  * Provenance and DocumentReference resources, which are not de-duplicated for attribution purposes.
```
Comment (au2) [inline README.md]: Meds also use organization identifier for deduplication

File: README.md [hunk 5]
```diff
@@ -66,8 +93,11 @@ This project incrementally builds and releases files for use in maven projects,
 ```
 
 
+## Transforming a CDA document to a Bundle of FHIR resources
+
+The below code is an annotated example of a basic CCD document transformation, further code examples can be found in [CCDTransformerTest.java](./src/test/java/tr/com/srdc/cda2fhir/CCDTransformerTest.java) file. You may also review all implemented interfaces in the [CCDTransformerImpl.java](./src/main/java/tr/com/srdc/cda2fhir/CCDTransformerImpl.java) file.
 
-## Transforming a CDA document to a Bundle of corresponding FHIR resources
+The output of this operation will be located at: `src/test/resources/output/C-CDA_R2-1_CCD-w-daf.json`.
```
Comment (au2) [inline README.md:102]: Didn't we get rid of daf profiles? Maybe refer to the one without the profile

**Review by au2 — CHANGES_REQUESTED** [review]
Inline.

File: README.md (hunk 1 above)
Comment (mmccall) [reply README.md:3]: fixed

File: README.md (hunk 2 above)
Comment (mmccall) [reply README.md]: added a note under the meds bullet.

File: README.md (hunk 3 above)
Comment (mmccall) [reply README.md:49]: added.

File: README.md (hunk 4 above)
Comment (mmccall) [reply README.md]: same as prior comment I believe?

File: README.md (hunk 5 above)
Comment (mmccall) [reply README.md:102]: yeah, removed all references to it.

**Comment by mmccall** [comment]
addressed afsin's feedback.

File: README.md (hunk 4 above)
Comment (au2) [reply README.md]: yes, it looks like there is nothing after ", and" here

File: README.md [hunk 6]
```diff
@@ -19,24 +19,51 @@ cda2fhir
 [![License Info](http://img.shields.io/badge/license-Apache%202.0-brightgreen.svg)](https://github.com/srdc/cda2fhir/blob/master/LICENSE.txt)
 [![Jenkins CI](https://jenkins.amida.com/buildStatus/icon?job=CDA2FHIR%20Tests/)](https://jenkins.amida.com/job/CDA2FHIR%20Tests/)
 
+##Overview
+cda2fhir is a Java library to transform HL7 [CDA R2](https://www.hl7.org/implement/standards/product_brief.cfm?product_id=7) instances to HL7 [FHIR](https://www.hl7.org/fhir/) resources. More specifically, cda2fhir enables automatic transformation of
```
Comment (au2) [inline README.md:23]: I think you need space after ##, it does not seem to be rendering

**Review by au2 — CHANGES_REQUESTED** [review]
I think you need space after all "##"'s and there is a unfinished sentence.  Marked inline

**Comment by mmccall** [comment]
> I think you need space after all "##"'s and there is a unfinished sentence. Marked inline

Gotcha, just fixed it and double checked it by looking on GH.

**Review by au2 — APPROVED** [review]

**Pull request merged**